File: src/rules/weak-password-hashing.py
import hashlib


def hash_password_md5(password):
    # ruleid: patronus-weak-password-hash-python
    return hashlib.md5(password).hexdigest()


def hash_pwd_sha256(pwd):
    # ruleid: patronus-weak-password-hash-python
    return hashlib.sha256(pwd).hexdigest()


def checksum(file_data):
    # ok: patronus-weak-password-hash-python
    return hashlib.sha256(file_data).hexdigest()

File: src/rules/weak-password-hashing.js
const crypto = require('crypto');

function hashPasswordMd5(password) {
  // ruleid: clipeus-weak-password-hash-js
  return crypto.createHash('md5').update(password).digest('hex');
}

function hashPwdSha1(pwd) {
  // ruleid: clipeus-weak-password-hash-js
  return crypto.createHash('sha1').update(pwd).digest('hex');
}

function hashUserPasswordSha256(userPassword) {
  // ruleid: clipeus-weak-password-hash-js
  return crypto.createHash('sha256').update(userPassword).digest('hex');
}

function checksumFile(fileData) {
  // ok: clipeus-weak-password-hash-js
  return crypto.createHash('sha256').update(fileData).digest('hex');
}
